### PR TITLE
Lcd print enhancements

### PR DESCRIPTION
**Pull request by dnlmlr**
### Added new print functions for ease of use on request of the h-ab lab:

`Ev3Printf`, `Ev3Println` and `Ev3Clear` are mostly wrappers for `LcdPrintf,` `LcdPrintln` and `LcdClean` which provide a more expected behavior.
`LcdPrintf` drops the color as the first parameter which is rarely used and causes confusions with the std `printf.`
`LcdPrintln` just adds a linefeed at the end.
`Ev3Clear` clears the screen and at the same time places the cursor back to the top left, which would previously require printing \f on top.

`TermPrintf` and `TermPrintln` work exactly like Ev3Printf and Ev3Println, but add the capability to automatically scroll upwards when the text would be out of bounds on the bottom of the screen. This works with all font sizes.

These are all packed into new separated functions to keep backwards compatibility, since changing the behavior of those functions might break running programs. 

Further changes include commenting a few more unused / not fully implemented functions, fixing minor bugs/typos and modifying the ev3_timer files. 

The ev3_timer module implements a callback structure by utilizing the linux alarm signal to throw an interrupt every 10 milliseconds. This interrupt is then used to execute initialized callback functions at a multiple of 10 milliseconds intervals.
Unfortunately this breaks any blocking os/system/kernel calls with interrupt. Since this is only used to refresh the display every 250 milliseconds, the option was added to disable the timer system by defining `DISABLE_TIMERS` and then refreshing the display manually after prints.
Additionally the option to change the raw timer interval was added and used to change the time from 10 milliseconds to 250 which is the only interval used by default anyways. This could even enhance the performance drastically in some situations.

Let me know if there are any unwanted changes or other (potential) problems.

### CODE REVIEW

File: API/ev3_lcd.h [hunk 1]
```diff
@@ -364,12 +364,33 @@ int asprintf(char **, const char *, ...);
...
 int Ev3Println(const char *fmt, ...);
+
+/**
+ * Clear the display and reset the cursor to the top left corner.
+ */
 void Ev3Clear();
 
+/**
+ * Analog to Ev3Printf but automatically scrolls upwards to prevent 
+ * text from beeing out of bounds.
```
Comment (a3f) [inline API/ev3_lcd.h]: s/beeing/being/

File: API/ev3_lcd.h [hunk 2]
```diff
@@ -364,12 +364,33 @@ int asprintf(char **, const char *, ...);
...
+ */
 int TermPrintf(const char *fmt, ...);
+
+/**
+ * Analog to Ev3Printf but automatically scrolls upwards to prevent 
+ * text from beeing out of bounds.
```
Comment (a3f) [inline API/ev3_lcd.h]: likewise.

File: API/ev3_lcd.c [hunk 3]
```diff
@@ -1847,3 +1847,38 @@ int LcdPrintf(char color, const char *fmt, ...)
...
+int Ev3Printf(const char *fmt, ...)
+{
+	va_list args;
+	char *buffer;
+
+	va_start(args, fmt);
+	vasprintf(&buffer, fmt, args);
+	va_end(args);
+
+	int ret = LcdPrintf(1, buffer);	// LcdPrintf with color=1 for black text
```
Comment (a3f) [inline API/ev3_lcd.c]: I would prefer if you added a LcdvPrintf which both `LcdPrintf` and `Ev3Printf` can use. This would avoid having the format string parsed twice.

You also don't `free` the memory allocated by vasprintf here. (I just noted LcdPrintf doesn't either...)

File: API/ev3_lcd.c [hunk 4]
```diff
@@ -1880,5 +1880,5 @@ int Ev3Println(const char *fmt, ...)
 void Ev3Clear()
 {
 	LcdClean();
-	LcdPrintf("\f");
```
Comment (a3f) [inline API/ev3_lcd.c]: How is that a typo? There is no necessity to use "%s" here.

File: API/ev3_lcd.c [hunk 5]
```diff
@@ -1688,30 +1672,52 @@ char RectOutEx(int x, int y, int width, int height, unsigned long options)
...
+		CmdDrawEllipse(x, y, radiusX, radiusY, pixelMode, fillMode);
 	return 0;
 }
 
-char NumOutEx(int x, int y, int value, unsigned long options)
+
+/**********************************************************************
+ * Unused / currently useless functions
+ */
+#if 0
```
Comment (a3f) [inline API/ev3_lcd.c]: Oh, I would just delete them. It's unlikely they will be ever implemented.

File: API/ev3_lcd.c [hunk 6]
```diff
@@ -1683,6 +1679,21 @@ char EllipseOutEx(int x, int y, uint8_t radiusX, uint8_t radiusY, unsigned long
 }
 
 
+/*******************
+ * Redundant functions
+ */
+#if 0
```
Comment (a3f) [inline API/ev3_lcd.c]: Why not leave it be?

File: API/ev3_lcd.c [hunk 7]
```diff
@@ -1986,5 +1988,5 @@ void TermPrintln(const char *fmt, ...)
 void Ev3Clear()
 {
 	LcdClean();
-	LcdPrintf("%s", "\f");
+	LcdPrintf(1, "%s", "\f");
```
Comment (a3f) [inline API/ev3_lcd.c]: Could you reorder the commits to squash these fixups?

File: API/ev3_lcd.c [hunk 8]
```diff
@@ -1883,17 +1892,95 @@ int Ev3Printf(const char *fmt, ...)
 
 int Ev3Println(const char *fmt, ...)
 {
-	char *buffer;
-	vaprintf(buffer, "%s%s", "\n");
```
Comment (a3f) [inline API/ev3_lcd.c]: Just a suggestion for the future, iIt would've been better if bugs were fixed in separate commits or even better squashed into the initial commit.

File: API/ev3_lcd.h [hunk 9]
```diff
@@ -58,6 +58,7 @@ extern "C" {
 #include <limits.h>
 #include <errno.h>
 #include <libgen.h>
+#include <stdarg.h>
```
Comment (a3f) [inline API/ev3_lcd.h]: Why is this needed for the header?

File: API/ev3_lcd.h (hunk 1 above)
Comment (dnlmlr) [reply API/ev3_lcd.h]: embarrassing and fixed

File: API/ev3_lcd.h (hunk 2 above)
Comment (dnlmlr) [reply API/ev3_lcd.h]: likewise

File: API/ev3_lcd.c (hunk 5 above)
Comment (dnlmlr) [reply API/ev3_lcd.c]: done

File: API/ev3_lcd.c (hunk 6 above)
Comment (dnlmlr) [reply API/ev3_lcd.c]: I changed it to be enabled by default

File: API/ev3_lcd.c (hunk 7 above)
Comment (dnlmlr) [reply API/ev3_lcd.c]: Good point, I squashed multiple commits that fixed bugs introduced by previous new commits.

File: API/ev3_lcd.h (hunk 9 above)
Comment (dnlmlr) [reply API/ev3_lcd.h]: It is not. I removed it.

File: API/ev3_lcd.c (hunk 4 above)
Comment (dnlmlr) [reply API/ev3_lcd.c]: I just checked. The read typo was the missing 1 as first argument. That is why the Ev3Printf is a good idea.
Since I just added the `"%s"` instead of fixing the actual issue, this commit is useless.

**Comment by a3f** [comment]
Sorry for the delay and thanks for the contribution.
